src/extremeweatherbench/utils.py

```diff
@@ -414,3 +431,175 @@ def read_event_yaml(input_pth: str | Path) -> dict:
     with open(input_pth, "rb") as f:
         yaml_event_case = yaml.safe_load(f)
     return yaml_event_case
+
+
+def unit_check(df: pd.DataFrame) -> pd.DataFrame:
+    """Base function to test for units in the point obs dataframe.
+    Potential to expand more comprehensively.
+
+    Arguments:
+        df: dataframe with point obs.
+
+
+    Returns a corrected dataframe."""
+    if "surface_air_temperature" in df.columns:
+        # Check if temperature appears to be in Kelvin (values mostly > 200)
+        if df["surface_air_temperature"].mean() > 200:
+            pass
+        else:
+            # Otherwise assume Celsius
+            df["surface_air_temperature"] = df["surface_air_temperature"] + 273.15
+    return df
```

Comment:
I don't think we need this - it could get endlessly complicated, and even in the temperature case here, the implementation is based purely on assumptions about the distribution of values in a column (which itself has an assumption about the column name).

It's far easier to rely on "garbage in, garbage out." Make explicit the requirements for forecast data and validating observations (e.g. units), and then let _caveat emptor_. It'll be obvious enough in the results.